### PR TITLE
Add GetPendingTransactions query to shared model

### PR DESCRIPTION
**Pull request by igor-egorov**
### Description of the Change

This PR adds an interface of GetPendingTransactions query to shared model. 
Currently, the query does NOT perform actual information retrieving (will be implemented after appearing of transactions batches support in develop branch).
Additionally, this pr provides documentation for the new query.

### Benefits

It is a starting point for further development of GetPendingTransactions feature.

### Possible Drawbacks 

None ?

### Usage Examples or Tests 

build everything and test everything, even bindings

### CODE REVIEW

File: irohad/execution/impl/query_execution.cpp [hunk 1]
```diff
@@ -332,6 +320,17 @@ QueryProcessingFactory::executeGetSignatories(
   auto response = QueryResponseBuilder().signatoriesResponse(*signs);
   return response;
 }
+
+QueryProcessingFactory::QueryResponseBuilderDone
+QueryProcessingFactory::executeGetPendingTransactions(
+    const shared_model::interface::GetPendingTransactions &query,
+    const shared_model::interface::types::AccountIdType &query_creator) {
+  std::vector<shared_model::proto::Transaction> txs;
```
Comment (l4l) [inline irohad/execution/impl/query_execution.cpp]: Always empty?

File: docs/source/api/queries.rst [hunk 2]
```diff
@@ -159,6 +159,44 @@ Response Structure
...
+    }
+
+Response Structure
+------------------
+
+.. csv-table::
+    :header: "Field", "Description", "Constraint", "Example"
+    :widths: 15, 30, 20, 15
+
+        "Transactions", "an array of transactions for given account", "Committed transactions", "{tx1, tx2…}"
```
[the remark this answers is not in the document]
Comment (kamilsa) [reply docs/source/api/queries.rst]: Or "list of unsigned transactions or batches"

File: irohad/execution/impl/query_execution.cpp (hunk 1 above)
Comment (kamilsa) [reply irohad/execution/impl/query_execution.cpp]: I think it is temporary unless query execution is implemented. Please leave a comment about that

File: docs/source/api/queries.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (l4l) [reply docs/source/api/queries.rst]: Unsigned is not really a good word here. Becuase [valid] multisig txes either signed or haven't been sent yet

File: irohad/execution/impl/query_execution.cpp (hunk 1 above)
Comment (l4l) [reply irohad/execution/impl/query_execution.cpp]: Sorry, haven't noticed that. Maybe you consider implementing mst? It's already implemented

File: irohad/execution/query_execution.hpp [hunk 3]
```diff
@@ -1,26 +1,16 @@
...
+ * SPDX-License-Identifier: Apache-2.0
  */
+
 #ifndef IROHA_QUERY_EXECUTION_HPP
 #define IROHA_QUERY_EXECUTION_HPP
 
 #include "ametsuchi/block_query.hpp"
 #include "ametsuchi/wsv_query.hpp"
-#include "builders/protobuf/builder_templates/query_response_template.hpp"
 #include "builders/protobuf/builder_templates/blocks_query_template.hpp"
```
[the remark this answers is not in the document]
Comment (igor-egorov) [reply irohad/execution/query_execution.hpp]: You are right. This was redundant.

File: irohad/execution/impl/query_execution.cpp (hunk 1 above)
Comment (igor-egorov) [reply irohad/execution/impl/query_execution.cpp]: It is ok. Now we have common understanding. TODO about missing implementation is placed on a line below.

File: docs/source/core_concepts/glossary.rst [hunk 4]
```diff
@@ -192,6 +182,16 @@ of signatures required to consider a transaction signed.
 The default value is 1.
 Each account can link additional public keys and increase own quorum number.
 
+Role
```
Comment (l4l) [inline docs/source/core_concepts/glossary.rst]: Could you please revert reordering, it differs from the pr aim

Comment (igor-egorov) [reply docs/source/core_concepts/glossary.rst]: Hmm. It was an intended change. Now the topics are sorted alphabetically if no other more valuable factors involved (e.g. logical composition).

File: docs/source/core_concepts/glossary.rst [hunk 5]
```diff
@@ -250,6 +258,48 @@ Transaction status set
...
+.. TODO igor-egorov, 2018-07-04, IR-1356, add here a link to MST docs
+
+Batch can contain transactions created by different accounts.
+Any transaction within a batch can require single or multiple signatures (depends on quorum set for an account of transaction creator).
+At least one transaction inside a batch should have at least one signature to let the batch pass `stateful validation`_.
+
+Atomic Batch
+------------
+
+All the transactions within an atomic batch should pass `stateful validation`_ for the batch to be applied to a ledger.
```
Comment (kamilsa) [inline docs/source/core_concepts/glossary.rst]: Please add sentence saying that order of transactions in atomic batch is preserved

**Pull request merged**